core/sched.c

```diff
@@ -100,21 +100,21 @@ static void _unschedule(thread_t *active_thread)
 #endif
 }
 
-int __attribute__((used)) sched_run(void)
+thread_t *__attribute__((used)) sched_run(void)
 {
+    sched_context_switch_request = 0;
```

Comment:
this is now done twice?

Reply:
That's a rebase mistake I think. see #14565, this should be cleared *after* the sleep.